src/include/network/network_io_utils.h

```diff
@@ -0,0 +1,426 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Terrier
+//
+// network_io_utils.h
+//
+// Identification: src/include/network/network_io_utils.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+#include <arpa/inet.h>
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "common/exception.h"
+#include "common/internal_types.h"
+#include "util/portable_endian.h"
+
+namespace terrier {
+namespace network {
+#define _CAST(type, val) ((type)(val))
```

Comment:
This looks pretty close to the buffer code we had to write for terrier. Again not that you have to do this in this PR, I think it might be interesting to look at the terrier code and explore the possibility of merging the two.